lib/utils/pcb-board-geometry.ts

```diff
@@ -241,6 +258,149 @@ const createHoleGeoms = (
+          const record = rotation as Record<string, unknown>
+          const degreeCandidate = [
+            record.deg,
+            record.degs,
+            record.degree,
+            record.degrees,
+            record.ccw,
+            record.ccw_degrees,
+            record.ccw_degree,
+          ].find(
+            (value): value is number =>
+              typeof value === "number" && Number.isFinite(value),
+          )
+          if (degreeCandidate !== undefined) {
+            rotationRad = (degreeCandidate * Math.PI) / 180
+          } else {
+            const radCandidate = [
+              record.rad,
+              record.rads,
+              record.radian,
+              record.radians,
+              record.ccw_radians,
+            ].find(
+              (value): value is number =>
+                typeof value === "number" && Number.isFinite(value),
+            )
+            if (radCandidate !== undefined) {
+              rotationRad = radCandidate
+            }
+          }
+        }
+        if (rotationRad) {
+          geom = rotateZ(-rotationRad, geom)
+        }
+
+        geoms.push(translate([relX, relY, 0], geom))
+        break
+      }
+      case "circle": {
+        const { center } = cutout
+        if (!center || !isFiniteNumber(center.x) || !isFiniteNumber(center.y)) {
+          continue
+        }
+
+        const radius = (() => {
+          if (
+            typeof cutout.radius === "number" &&
+            Number.isFinite(cutout.radius)
+          ) {
+            return cutout.radius
+          }
+
+          if (
+            "diameter" in cutout &&
+            typeof cutout.diameter === "number" &&
+            Number.isFinite(cutout.diameter)
+          ) {
+            return cutout.diameter / 2
+          }
+
+          return undefined
+        })()
+
+        if (!radius) continue
+
+        const relX = center.x - boardCenter.x
+        const relY = -(center.y - boardCenter.y)
+
+        geoms.push(createCircularHole(relX, relY, radius, thickness))
+        break
+      }
+      case "polygon": {
+        const { points } = cutout
+        if (!Array.isArray(points) || points.length < 3) continue
+
+        let polygonPoints = points
+          .filter(
+            (point): point is Point =>
+              point !== undefined &&
+              isFiniteNumber(point.x) &&
+              isFiniteNumber(point.y),
+          )
+          .map((point) => toBoardSpaceVec2(point, boardCenter))
+
+        if (polygonPoints.length < 3) continue
+
+        if (arePointsClockwise(polygonPoints)) {
+          polygonPoints = polygonPoints.slice().reverse()
+        }
+
+        const polygon2d = polygon({ points: polygonPoints })
+        let geom = extrudeLinear({ height: thickness + 1 }, polygon2d)
+        geom = translate([0, 0, -(thickness + 1) / 2], geom)
+        geoms.push(geom)
+        break
+      }
+      default:
+        break
+    }
+  }
+
+  return geoms
+}
+
```

Comment:
can you move this function to a separate file? It's not good to have files over 300 lines